src/pages/studyView/charts/categoryBarChart/CategoryBarChart.tsx

```diff
@@ -0,0 +1,353 @@
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import {
+    VictoryAxis,
+    VictoryBar,
+    VictoryChart,
+    VictoryLabel,
+    VictorySelectionContainer,
+} from 'victory';
+import { computed, observable, makeObservable } from 'mobx';
+import _ from 'lodash';
+import { DataFilterValue } from 'cbioportal-ts-api-client';
+import { AbstractChart } from 'pages/studyView/charts/ChartContainer';
+import autobind from 'autobind-decorator';
+import CategoryBarChartAxisLabel from './CategoryBarChartAxisLabel';
+import {
+    filterCategoryBins,
+    clinicalDataToDataBin,
+    generateCategoricalData,
+    DataBin,
+    isDataBinSelected,
+    ClinicalDataCountSummary,
+} from '../../StudyViewUtils';
+import { STUDY_VIEW_CONFIG } from '../../StudyViewConfig';
+import { DEFAULT_NA_COLOR } from 'shared/lib/Colors';
+import CategoryBarChartToolTip, {
+    ToolTipModel,
+} from './CategoryBarChartToolTip';
+import WindowStore from 'shared/components/window/WindowStore';
+import ReactDOM from 'react-dom';
+import {
+    CBIOPORTAL_VICTORY_THEME,
+    calcPlotBottomPadding,
+} from 'cbioportal-frontend-commons';
+
+export interface IBarChartProps {
+    data: ClinicalDataCountSummary[];
+    width: number;
+    height: number;
+    filters: DataFilterValue[];
+    onUserSelection: (dataBins: DataBin[]) => void;
+    showNAChecked: boolean;
+}
+
+export type BarDatum = {
+    x: number;
+    y: number;
+    dataBin: DataBin;
+};
+
+function generateTheme() {
+    const theme = _.cloneDeep(CBIOPORTAL_VICTORY_THEME);
+    theme.axis.style.tickLabels.fontSize *= 0.85;
+
+    return theme;
+}
+
+const VICTORY_THEME = generateTheme();
+const TILT_ANGLE = 30;
+
+@observer
+export default class CategoryBarChart
```

Comment:
i'm noticing some similarities with our existing BarChart component. can we look into maybe extending it or reusing that code to prevent duplicate code?